SharkCage/CageServiceInstaller/ServiceInstaller.cs

```diff
@@ -0,0 +1,391 @@
+
+                if (service == IntPtr.Zero)
+                    throw new ApplicationException("Failed to install service.");
+
+                try
+                {
+                    StartService(service);
+                }
+                finally
+                {
+                    CloseServiceHandle(service);
+                }
+            }
+            finally
+            {
+                CloseServiceHandle(scm);
+            }
+        }
+
+        public static void StartService(string service_name)
+        {
+            IntPtr scm = OpenSCManager(ScmAccessRights.Connect);
+
+            try
+            {
+                IntPtr service = OpenService(scm, service_name, ServiceAccessRights.QueryStatus | ServiceAccessRights.Start);
+                if (service == IntPtr.Zero)
+                    throw new ApplicationException("Could not open service.");
+
+                try
+                {
+                    StartService(service);
+                }
+                finally
+                {
+                    CloseServiceHandle(service);
+                }
+            }
+            finally
+            {
+                CloseServiceHandle(scm);
+            }
+        }
+
+        public static void StopService(string service_name)
+        {
+            IntPtr scm = OpenSCManager(ScmAccessRights.Connect);
+
+            try
+            {
+                IntPtr service = OpenService(scm, service_name, ServiceAccessRights.QueryStatus | ServiceAccessRights.Stop);
+                if (service == IntPtr.Zero)
+                    throw new ApplicationException("Could not open service.");
+
+                try
+                {
+                    StopService(service);
+                }
+                finally
+                {
+                    CloseServiceHandle(service);
+                }
+            }
+            finally
+            {
+                CloseServiceHandle(scm);
+            }
+        }
+
+        private static void StartService(IntPtr service)
+        {
+            SERVICE_STATUS status = new SERVICE_STATUS();
+            StartService(service, 0, 0);
+            var changedStatus = WaitForServiceStatus(service, ServiceState.StartPending, ServiceState.Running);
+            if (!changedStatus)
+                throw new ApplicationException("Unable to start service");
+        }
+
+        private static void StopService(IntPtr service)
+        {
+            SERVICE_STATUS status = new SERVICE_STATUS();
+            ControlService(service, ServiceControl.Stop, status);
+
+            var amount_of_retries = 5;
+            var changed_status = false;
+
+            // It takes sometimes longer to stop the service than the dwWaitHint is set by the system.
+            for (; amount_of_retries > 0; amount_of_retries--)
```

Comment:
is there a reason `amount_of_retries` is not initialized in the for loop directly?